### PR TITLE
Pytest: run tests in random order

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Should be good after #30151

File: conftest.py [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+
   # setup a clean environment for each test
   with OpenpilotPrefix():
     yield
+
+  os.environ.clear()
+  os.environ.update(starting_env)
+
+
+@pytest.fixture(scope="class", autouse=True)
```
Comment (adeebshihadeh) [inline conftest.py:19]: why do we need two fixtures for this?

Comment (jnewb1) [reply conftest.py:19]: if you use setUpClass, the environment variable will persist

Comment (jnewb1) [reply conftest.py:19]: so you need to clear the environment variables before each test function, and each class

Comment (adeebshihadeh) [reply conftest.py:19]: ah, can you leave a comment explaining this? it's not obvious from reading this

**Comment by jnewb1** [comment]
> Should be good after #30151

still seems like an issue https://jenkins.comma.life/blue/organizations/jenkins/openpilot/detail/random-order/21/pipeline/

File: conftest.py (hunk 1 above)
Comment (jnewb1) [reply conftest.py:21]: added a comment

**Pull request merged**